lightning/src/ln/peer_handler.rs

```diff
@@ -343,14 +371,16 @@ pub type SimpleRefPeerManager<'a, 'b, 'c, 'd, 'e, 'f, 'g, SD, M, T, F, C, L> = P
 /// you're using lightning-net-tokio.
 ///
 /// [`read_event`]: PeerManager::read_event
-pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> where
+pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> where
 		CM::Target: ChannelMessageHandler,
 		RM::Target: RoutingMessageHandler,
-		L::Target: Logger {
+		L::Target: Logger,
+		CMH::Target: CustomMessageHandler + wire::CustomMessageReader {
```

Comment:
nit: you dont need the extra bound here since `CustomMessageHandler: wire::CustomMessageReader`

Reply:
Indeed, fixed